contrib/init/systemd/docker.service

```diff
@@ -1,7 +1,7 @@
 [Unit]
 Description=Docker Application Container Engine
 Documentation=https://docs.docker.com
-After=network.target docker.socket firewalld.service
+After=network-online.target docker.socket firewalld.service
```

Comment:
Following the discussion in https://github.com/docker/docker/issues/29876#issue-198677228, @JamieGarside suggests to also add;

```
Wants=network-online.target
```